packages/react/src/time-input/utils.test.ts

```diff
@@ -0,0 +1,162 @@
+		expect(isValidTime('66666')).toEqual(false);
+		expect(isValidTime('666666')).toEqual(false);
+		expect(isValidTime('25')).toEqual(false);
+		expect(isValidTime('2500')).toEqual(false);
+		expect(isValidTime('25:00')).toEqual(false);
+		expect(isValidTime('9:66')).toEqual(false);
+		expect(isValidTime('pm')).toEqual(false);
+		expect(isValidTime('am')).toEqual(false);
+	});
+});
+
+describe('formatTime', () => {
+	test('Single or double-digits less than 12 return time in am', () => {
+		for (let i = 1; i < 12; i += 1) {
+			expect(formatTime(`${i}`, 'h:mm aaa')).toEqual(`${i}:00 am`);
+			expect(formatTime(`${i}`, 'hh:mm aaa')).toEqual(
+				`${i}:00 am`.padStart(8, '0')
+			);
+			expect(formatTime(`${i}`, 'HH:mm')).toEqual(`${i}:00`.padStart(5, '0'));
+		}
+	});
+
+	test('0 returns 12:00 am', () => {
+		expect(formatTime('0', 'h:mm aaa')).toEqual('12:00 am');
+		expect(formatTime('0', 'hh:mm aaa')).toEqual('12:00 am');
+		expect(formatTime('0', 'HH:mm')).toEqual('00:00');
+	});
+
+	test('Single or double-digits greater than 12 return time in pm', () => {
+		for (let i = 13; i < 24; i += 1) {
+			expect(formatTime(`${i}`, 'h:mm aaa')).toEqual(`${i - 12}:00 pm`);
+			expect(formatTime(`${i}`, 'hh:mm aaa')).toEqual(
+				`${i - 12}:00 pm`.padStart(8, '0')
+			);
+
+			expect(formatTime(`${i}`, 'HH:mm')).toEqual(`${i}:00`.padStart(5, '0'));
+		}
+	});
+
+	test('12 returns 12:00 pm', () => {
+		expect(formatTime('12', 'h:mm aaa')).toEqual('12:00 pm');
+		expect(formatTime('12', 'hh:mm aaa')).toEqual('12:00 pm');
+		expect(formatTime('12', 'HH:mm')).toEqual('12:00');
+	});
+
+	test('24 returns 12:00 am', () => {
+		expect(formatTime('24', 'h:mm aaa')).toEqual('12:00 am');
+		expect(formatTime('24', 'hh:mm aaa')).toEqual('12:00 am');
+		expect(formatTime('24', 'HH:mm')).toEqual('00:00');
+	});
+
+	test('3-digit numbers provide minutes', () => {
+		expect(formatTime('115', 'h:mm aaa')).toEqual('1:15 am');
+		expect(formatTime('230', 'h:mm aaa')).toEqual('2:30 am');
+		expect(formatTime('345', 'h:mm aaa')).toEqual('3:45 am');
+	});
+
+	test('Increments of 100 up to 2400 provide the time on the hour', () => {
+		for (let i = 1; i < 12; i += 1) {
+			expect(formatTime(`${i * 100}`, 'h:mm aaa')).toEqual(`${i}:00 am`);
+		}
+		expect(formatTime('1200', 'h:mm aaa')).toEqual('12:00 pm');
+		for (let i = 13; i < 24; i += 1) {
+			expect(formatTime(`${i * 100}`, 'h:mm aaa')).toEqual(`${i - 12}:00 pm`);
+		}
+		expect(formatTime('2400', 'h:mm aaa')).toEqual('12:00 am');
+	});
+
+	test('4-digit numbers', () => {
+		expect(formatTime('0115', 'h:mm aaa')).toEqual('1:15 am');
+		expect(formatTime('0230', 'h:mm aaa')).toEqual('2:30 am');
+		expect(formatTime('0345', 'h:mm aaa')).toEqual('3:45 am');
+		expect(formatTime('1315', 'h:mm aaa')).toEqual('1:15 pm');
+		expect(formatTime('1430', 'h:mm aaa')).toEqual('2:30 pm');
+		expect(formatTime('1545', 'h:mm aaa')).toEqual('3:45 pm');
+	});
+
+	test('5-digit numbers', () => {
```

Comment:
These are 6-digit numbers

```suggestion
	test('6-digit numbers', () => {
```

Reply:
lol good pickup